udmis/src/main/java/com/google/bos/udmi/monitoring/LogTail.java

```diff
@@ -0,0 +1,253 @@
+package com.google.bos.udmi.monitoring;
```

Comment:
seems like this class is trying to do two things?  One is gather metrics, the other is emit them -- I'd separate them out so they can be repurposed (e.g. emit other sources of metrics, or gather metrics to a different sink).

Reply:
Yeah, this was just to get started i.e. draft PR, I split this into pieces.